custom_components/waste_collection_schedule/waste_collection_schedule/source/pronatura_bydgoszcz.py

```diff
@@ -0,0 +1,138 @@
+    "odpady wielkogabarytowe": "mdi:wardrobe",
+}
+
+class Source:
+    def __init__(self, city, street_name, street_number, lang="pl"):
+        self._city = city.upper()
+        self._street_name = street_name.upper()
+        self._street_number = str(street_number).upper()
+        self._lang = lang if lang in ["pl", "en"] else "pl"  # Fallback to 'pl' if not 'pl' or 'en'
+
+    def fetch(self):
+        try:
+            return self.get_data()
+        except Exception as e:
+            _LOGGER.error(f"fetch failed for source {TITLE}: {e}")
+            return []
+
+    def get_data(self):
+        streets_url = f"{URL}/streets"
+        r = requests.get(streets_url)
+        r.raise_for_status()
+        streets = json.loads(r.text)
+        street_id = None
+        for street in streets:
+            if street["street"].upper() == self._street_name:
+                street_id = street["id"]
+                break
+        if street_id is None:
+            raise Exception("Street not found")
+
+        addresses_url = f"{URL}/address-points/{street_id}"
+        r = requests.get(addresses_url)
+        r.raise_for_status()
+        addresses = json.loads(r.text)
+        address_id = None
+        for address in addresses:
+            if address["buildingNumber"].upper() == self._street_number:
+                address_id = address["id"]
+                break
+        if address_id is None:
+            raise Exception("Address not found")
+
+        schedule_url = f"{URL}/trash-schedule/{address_id}"
+        r = requests.get(schedule_url)
+        r.raise_for_status()
+        schedule = json.loads(r.text)
+        trash_schedule = schedule["trashSchedule"]
+
+        entries = []
+        year = schedule["year"]
+        name_map = NAME_MAP_EN if self._lang == "en" else NAME_MAP_PL
+        for month_schedule in trash_schedule:
+            month = self._get_month_number(month_schedule["month"])
+            # Collect recykling days
+            recykling_days = set()
+            for collection in month_schedule["schedule"]:
+                waste_type = collection["type"]
+                for day in collection["days"]:
+                    if waste_type in ["papier", "plastik", "szkło"]:
+                        recykling_days.add(int(day))
+                    else:
+                        entries.append(
+                            Collection(
+                                datetime.date(year, month, int(day)),
+                                name_map[waste_type],
+                                ICON_MAP[waste_type],
+                            )
+                        )
+            # Add recykling entries
+            for day in recykling_days:
+                entries.append(
+                    Collection(
+                        datetime.date(year, month, day),
+                        name_map["papier"],
+                        "mdi:recycle",
+                    )
+                )
```

Comment:
Is there a reason you are mapping paper plastic and glass to recycling instead of showing all 3. Are they always at the same date? For all addresses?

Reply:
i will revert to separate because im not sure its all addresses, and that merge is not really necessary